### ISSUE TITLE
[linux] Detached subprocesses should be independent of lf

### ISSUE THREAD

**Issue by slavistan** [gokcehan/lf#694]
lf spawns subprocesses created by`shell-pipe`, `shell-async` etc. within its own process group. This holds true even if the subprocess is detached from lf's process tree when the command is executed in the background.

Consider the following example. Running the command `&zathura &` from lf will yield the follow process structure:

```
┌ runit(1,1)
├───st(PID: 18099, PGID: 18099)───zsh(PID 18100, PGID 18100)───lf(PID 18251, PGID 18251) 
└───zathura(PID 18654, PGID 18251)
```

Note that zathura and lf share the same process group. This will cause zathura, the program spawned by lf, to crash if the terminal is closed. You can verify this yourself by opening a terminal and starting lf, running a background process via `&zathura &` and then `xkill`ing the terminal. I'm not aware of all the details but I think the kernel sends a SIGHUP to every child and its group.

I find this behavior surprising and unintuitive. I often spawn gui programs and silent background processes from lf and leave lf running in the terminal for later usage. Then, at some later point I close my terminal  which unintentionally kills my application (zathura or gimp usually). It really took a while to find out what's going on, so here's my best effort to save somebody else some time in the future :-)

My suggestions are:
1. `shell-async` commands should run in a separate process group.
2. All types of shell commands should run in a separate process group and subprocess communication should be performed via signals (if that's even required).
3. All existing shell commands are left as they are and an additional shell command construct is introduced which spawns all subprocesses in separate process groups. (Maybe just repeat each prefix character? `!!`, `&&` etc.) 
4. An explicit warning should be added to the documentation, including a workaround (e.g. `&setsid --fork zathura &`).

Kind regards.

P.S.: vim has the same quirk (same process group, even if detached). `:!zathura &` will crash zathura if the terminal is killed. Neovim cleaned that up and spawns each subprocess in its own process group (suggestion 2 above).

**Comment by gokcehan** [comment]
@slavistan Thanks for the proposal and sorry for late reply.

It sounds like an annoying behavior. Unfortunately, I can't reproduce it on my linux machine. It is most likely due to different init systems we are using. I have an Arch linux with systemd and you seem to be using a system with runit.

Are there any disadvatages for running commands in a separate group? What do you mean by subprocess communication? If there are no disadvantages, then (3) may not be necessary especially since we already have too many different command types. (1) and (2) should be fine but these things might be hard to implement. I don't see anything related to this in Go standard library so one should essentially test this on various platforms (e.g. Windows, Mac, BSD, and Linux with various init systems).

Regarding (4), we already have a note about the use of `setsid` in "Opening Files" section in the documentation. We could add something more about `setsid --fork` if desired.

**Comment by slavistan** [comment]
Speaking of late replies - here I am two months later 🙂

Do the following instructions leave you with a running instance of zathura? This is a shell one-liner equivalent of the scenario I asked you to try to reproduce in my first post.

```
( sleep 2 && xdotool windowkill $WINDOWID )& lf -command "&zathura"
```
I ran this on an ubuntu, an arch and an artix machine and Zathura shuts down every time the terminal window gets killed by xdotool. May I kindly ask you to tell me what the behavior on your machine is?

**Comment by gokcehan** [comment]
@slavistan Yes I can reproduce the issue with that command now. I can also reproduce the issue as you describe in your original post. I have been using `mupdf` for pdf files so maybe I was testing it with `mupdf` instead of `zathura` before. Note, `mupdf` does not have this issue for some reason. In any case, I'm open to solutions for this issue but I don't think I have spare time to work on this myself and any PR should ideally be tested on various platforms.

**Comment by slavistan** [comment]
Fixed by #753.

@gokcehan Should this issue be closed or should we check the other platforms? I have a Windows 11 machine where I could perform a surface-level check (opening a program from lf and killing the terminal from the task manager). My understanding of the inner workings of Windows is rather limited, however.

Oh, and on my machine, `mupdf` behaves just like `zathura` and used to crash before #753, when lf's terminal was killed ungracefully.

**Comment by gokcehan** [comment]
@slavistan I have tried a few simple scenarios in Windows and did not see any issues. I think we can close this until someone comes up with a way to reproduce it in some platform.
